src/frontend/sharedComponents/DeviceCard.tsx

```diff
@@ -24,16 +29,29 @@ export const DeviceCard = ({
   deviceId,
   dateAdded,
   onPress,
-  isConnected = true,
+  deviceConnectionStatus,
 }: DeviceCardProps) => {
+  let isDisconnected: boolean;
+  switch (deviceConnectionStatus) {
+    case undefined:
+    case 'connected':
+      isDisconnected = false;
+      break;
+    case 'disconnected':
+      isDisconnected = true;
+      break;
+    default:
+      throw new ExhaustivenessError(deviceConnectionStatus);
+  }
+
   return (
     <TouchableOpacity
-      disabled={!onPress || !isConnected}
+      disabled={!onPress || isDisconnected}
```

Comment:
nit: seems sufficient to do `!onPress || deviceConnectionStatus === 'disconnected'` in this case